doc/api/pyplot_summary.rst

```diff
@@ -1,29 +1,283 @@
+    :nosignatures:
 
-   pyplot
+    colormaps
+    get_plot_commands
 
+Global functions
+================
 
-.. currentmodule:: matplotlib.pyplot
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-.. autofunction:: plotting
+    findobj
+    install_repl_displayhook
+    ioff
+    ion
+    isinteractive
+    pause
+    rc
+    rc_context
+    rcdefaults
+    show
+    switch_backend
+    uninstall_repl_displayhook
 
+Any artist
+==========
 
-Colors in Matplotlib
---------------------
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-There are many colormaps you can use to map data onto color values.
-Below we list several ways in which color can be utilized in Matplotlib.
+    getp
+    setp
+    xkcd
 
-For a more in-depth look at colormaps, see the
-:doc:`/tutorials/colors/colormaps` tutorial.
+Making figures
+==============
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-.. currentmodule:: matplotlib.pyplot
+    figure
+    subplots
+
+Working with figures
+====================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    clf
+    close
+    connect
+    disconnect
+    draw
+    fignum_exists
+    figure
+    gcf
+    get_current_fig_manager
+    get_figlabels
+    get_fignums
+    ginput
+    savefig
+    waitforbuttonpress
+
+Putting things in figures
+=========================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    figimage
+    figlegend
+    figtext
+    suptitle
+
+Working with axes
+=================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    axes
+    box
+    cla
+    delaxes
+    gca
+    grid
+    sca
+    subplot
+    subplots
+    subplot2grid
+    subplot_tool
+    subplots_adjust
+    tight_layout
+    twinx
+    twiny
+
+Working with axis
+=================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    autoscale
+    axis
+    locator_params
+    margins
+    minorticks_off
+    minorticks_on
+    rgrids
+    thetagrids
+    tick_params
+    ticklabel_format
+    xlabel
+    ylabel
+    xlim
+    ylim
+    xscale
+    yscale
+    xticks
+    yticks
+
+
+Current image
+=============
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    clabel
+    clim
+    colorbar
+    gci
+    sci
+    set_cmap
+
+Putting things in axes
+======================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    annotate
+    arrow
+    legend
+    table
+    text
+    title
+
+Working with images
+===================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    imread
+    imsave
+
+Plotting functions
```

Comment:
Might be worth moving this up too, since these are probably the second most common set of functions that people use.